.github/workflows/build-test.yml

```diff
@@ -77,3 +77,24 @@ jobs:
         run: |
           cd $GITHUB_WORKSPACE/build
           ../tools/run_all_tests.sh disable-timing-tests
+
+ # This job builds & runs iceoryx tests in Windows 2019
+  build-test-windows:
+    runs-on: windows-latest
+    # Softwares installed in windows instance
+    # https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name : Checkout
+        uses: actions/checkout@v2
+
+      - name: Build sources
+        run: |
+          cmake -Bbuild -Hiceoryx_meta -Dtest=ON && cmake --build build
+
+      - name: Run tests
+      # at the moment we do not support windows therefore we just run a subset of all the tests
+        run: |
+          build\utils\test\Debug\utils_moduletests.exe --gtest_filter="Semaphore*"
```

Comment:
We need to ensure full windows compability and enable the other tests too. I created a ticket for this: https://github.com/eclipse/iceoryx/issues/185